File: 2025/problems/gestalisti/submissions/accepted/runar_100.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <unordered_set>

using namespace std;

int main() {
    int n;
    cin >> n;
    unordered_set<string> guestBook;

    for (int i = 0; i < n; ++i) {
        char operation;
        cin >> operation;

        string name;
        getline(cin, name);

        if (operation == '+') {
            guestBook.insert(name);
        } else if (operation == '-') {
            guestBook.erase(name);
        } else if (operation == '?') {
            cout << (guestBook.find(name) != guestBook.end() ? "Jebb"
                                                             : "Neibb");
            cout << endl;
        }
    }

    return 0;
}

File: 2025/problems/gestalisti/submissions/accepted/runar_100.py
#!/usr/bin/python3
def main():
    n = int(input())
    guest_book = set()
    for _ in range(n):
        operation, name = input().split(maxsplit=1)
        if operation == "+":
            guest_book.add(name)
        elif operation == "-":
            try:
                guest_book.remove(name)
            except KeyError:
                pass
        elif operation == "?":
            print("Jebb" if name in guest_book else "Neibb")


if __name__ == "__main__":
    main()
